### PR TITLE
Improve logging in minio tests

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add logging for mock HTTP servers used in minio integration tests.

Follow-up to https://github.com/ClickHouse/ClickHouse/pull/70786

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 5dc443d5f61a2be7c28924a5d45f01757feb3eda with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70943/5dc443d5f61a2be7c28924a5d45f01757feb3eda/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70938/ff191b0b089af03d2134dd1f2bedd390cf42482f/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/stateless_tests__debug__[1_2].html">✅ success</a></td></tr>
<tr><td>Stress test</td><td>Runs stateless functional tests concurrently from several clients to detect concurrency-related errors</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/stress_test__debug_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91223fc4eb6c1da1e7cdf61fb28f9c70183c4386/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/integration/compose/docker_compose_minio.yml [hunk 1]
```diff
@@ -39,6 +39,10 @@ services:
     depends_on:
       - proxy1
       - proxy2
+    volumes:
+      - type: ${RESOLVER_LOGS_FS:-tmpfs}
+        source: ${RESOLVER_LOGS:-}
+        target: /var/log/resolver
```
Comment (vitlibar) [inline tests/integration/compose/docker_compose_minio.yml:45]: Some tests run scripts like [unstable_proxy.py](https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_merge_tree_s3/s3_mocks/unstable_proxy.py) so it's better to be able to see the results of those tests too. So after this PR we'll see such results in a folder named `_instances/resolver`

File: tests/integration/test_merge_tree_s3/s3_mocks/unstable_proxy.py [hunk 2]
```diff
@@ -70,7 +70,6 @@ def do_HEAD(self):
         if random.random() < 0.25 and len(r.content) > 1024 * 1024:
             r.content = r.content[: len(r.content) // 2]
         self.wfile.write(r.content)
-        self.wfile.close()
```
Comment (vitlibar) [inline tests/integration/test_merge_tree_s3/s3_mocks/unstable_proxy.py:73]: This change is not related to the added logging - it's only to remove "ValueError: I/O operation on closed file" in "python/http/server.py" from logs when running tests.

File: tests/integration/helpers/cluster.py [hunk 3]
```diff
@@ -2997,6 +3000,7 @@ def logging_pulling_images(**kwargs):
                 os.mkdir(self.minio_dir)
                 if self.minio_certs_dir is None:
                     os.mkdir(os.path.join(self.minio_dir, "certs"))
+                    os.mkdir(os.path.join(self.minio_dir, "certs", "CAs"))
```
Comment (vitlibar) [inline tests/integration/helpers/cluster.py:3003]: This change isn't related to the added logging - it's just to ensure that the `CAs` folder is created with the privileges which allow removing that folder on a repeated test run.

**Review by qoega — APPROVED** [review]

**Comment by qoega** [comment]
I checked https://s3.amazonaws.com/clickhouse-test-reports/70943/5cf74f4da2d71bf971d720c6a38bc00e00c3cf36/integration_tests__asan__old_analyzer__[1_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel4_0.tar.zst


```
cat ./_instances-0-gw0/resolver/no_list_objects.err.log
```

```
127.0.0.1 - - [22/Oct/2024 19:26:44] "GET / HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:28:52] "PUT /root/test_no_list_6.csv HTTP/1.1" 200 -
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 57854)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
172.16.5.7 - - [22/Oct/2024 19:28:52] "PUT /root/test_no_list_12.csv HTTP/1.1" 200 -
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 425, in handle_one_request
    self.wfile.flush() #actually send the response if not already done.
    ^^^^^^^^^^^^^^^^^^
ValueError: I/O operation on closed file.
----------------------------------------
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 57868)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 425, in handle_one_request
    self.wfile.flush() #actually send the response if not already done.
    ^^^^^^^^^^^^^^^^^^
ValueError: I/O operation on closed file.
----------------------------------------
```

```
cat ./_instances-0-gw0/resolver/unstable_server.err.log
127.0.0.1 - - [22/Oct/2024 19:26:46] "GET / HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:30:45] "HEAD /root/slow_send_test.csv HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:30:45] "GET /root/slow_send_test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:30:46] "GET /root/slow_send_test.csv HTTP/1.1" 206 -
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 39766)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 424, in handle_one_request
    method()
  File "//unstable_server.py", line 111, in do_GET
    self.wfile.write(
  File "/usr/local/lib/python3.12/socketserver.py", line 840, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
172.16.5.7 - - [22/Oct/2024 19:34:10] "HEAD /root/test.csv HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
```

in test_storage_s3 for example.

**Comment by vitlibar** [comment]
> ValueError: I/O operation on closed file.

I fixed that:
https://github.com/ClickHouse/ClickHouse/pull/70943/commits/871b938dfc743c7005a608ed2bb27cd3385d26ca
